File: 2015/1.lua
#!/usr/bin/env lua

--[[
  Advent of Code 2015 day 1 solution in Lua
  http://adventofcode.com/2015/day/1
  Lloyd Dilley
]]

UP = '('            -- instruction to go up a floor
DOWN = ')'          -- instruction to go down a floor
FILE_NAME = "1.dat" -- data file name

position = 0        -- current floor number
data_file, err = io.open(FILE_NAME, "r")
if not data_file then
  print("Unable to parse input file: " .. err)
end
repeat
  local direction = data_file:read(1)
  if direction == UP then
    position = position + 1
  elseif direction == DOWN then
    position = position - 1
  elseif direction == nil then
    break
  end
  print(string.format("Current floor: #%d", position))
until direction == nil
data_file:close()
print(string.format("Arrived at destination floor: #%d", position))

File: 2015/1.jl
# Advent of Code 2015 day 1 solution in Julia
# http://adventofcode.com/2015/day/1
# Lloyd Dilley

const UP = '('            # instruction to go up a floor
const DOWN = ')'          # instruction to go down a floor
const FILE_NAME = "1.dat" # data file name

try
  position = 0            # current floor number
  data_file = open(FILE_NAME, "r")
  data = read(data_file, String)
  close(data_file)
  for direction in data
    if direction == UP
      position += 1
    end
    if direction == DOWN
      position -= 1
    end
    println("Current floor: #$position")
  end
  println("Arrived at destination floor: #$position")
catch e
  println("Unable to parse input file: $e")
end